pkg/defaults/defaults.go

```diff
@@ -363,18 +363,26 @@ func fromConfig(
 		if config.PromotionConfiguration == nil {
 			return nil, nil, fmt.Errorf("cannot promote images, no promotion configuration defined")
 		}
-		postSteps = append(postSteps, releasesteps.PromotionStep(api.PromotionStepName, config, requiredNames, jobSpec, podClient, pushSecret, registryDomain(config.PromotionConfiguration), api.DefaultMirrorFunc))
+		registry := registryDomain(config.PromotionConfiguration)
+		postSteps = append(postSteps, releasesteps.PromotionStep(api.PromotionStepName, config, requiredNames, jobSpec, podClient, pushSecret, registry, targetName(registry, *config.PromotionConfiguration, "/", "/", ":"), api.DefaultMirrorFunc))
 		// Used primarily (only?) by the ci-chat-bot
 		if config.PromotionConfiguration.RegistryOverride != "" {
 			logrus.Info("No images to promote to quay.io if the registry is overridden")
 		} else {
-			postSteps = append(postSteps, releasesteps.PromotionStep(api.PromotionQuayStepName, config, requiredNames, jobSpec, podClient, pushSecret, api.QuayOpenShiftCIRepo, api.QuayMirrorFunc))
+			postSteps = append(postSteps, releasesteps.PromotionStep(api.PromotionQuayStepName, config, requiredNames, jobSpec, podClient, pushSecret, api.QuayOpenShiftCIRepo, targetName(api.QuayOpenShiftCIRepo, *config.PromotionConfiguration, ":", "_", "_"), api.QuayMirrorFunc))
 		}
 	}
 
 	return append(overridableSteps, buildSteps...), postSteps, nil
 }
 
+func targetName(registry string, config api.PromotionConfiguration, d1, d2, d3 string) string {
```

Comment:
d1, d2, d3 are ugly.
But I do not find another way provided you do not want me to do the following check

```go
if registry==... {
}
```

Reply:
Just make it two functions.